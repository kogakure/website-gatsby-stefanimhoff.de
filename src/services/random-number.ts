export const getRandomNumber = (max: number) =>
  String(Math.floor(Math.random() * Math.floor(max)));
