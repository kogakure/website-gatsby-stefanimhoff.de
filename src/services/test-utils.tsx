import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
