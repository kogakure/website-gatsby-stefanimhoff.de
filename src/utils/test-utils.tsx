import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
