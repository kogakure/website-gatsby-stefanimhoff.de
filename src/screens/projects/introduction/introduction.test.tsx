import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../../services/test-utils';

import { Introduction } from '.';

describe('Projects', () => {
  describe('Introduction', () => {
    test('renders correctly', () => {
      const { container } = render(<Introduction />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<Introduction />);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
