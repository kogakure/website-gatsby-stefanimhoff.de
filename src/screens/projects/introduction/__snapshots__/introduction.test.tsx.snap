// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Introduction renders correctly 1`] = `
.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.6;
  margin-bottom: clamp(1.44rem,3.70vw,1.728rem);
  margin-top: 0;
}

.c2 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c2:hover,
.c2:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c0 {
  grid-column: 2 / -2;
  grid-row: 2;
}

@media (min-width:768px) {
  .c0 {
    grid-column: 6 / -6;
  }
}

<div
  class="c0"
>
  <p
    class="c1"
  >
    Iâ€™m a 
    <em>
      Frontend Developer
    </em>
     by profession. But I worked in previous jobs also in Design, Editing, or 3D Animation. I even created Flash Animations and Multimedia CD-ROMs when this was still a thing.
  </p>
  <p
    class="c1"
  >
    I like the concept of the
     
    <a
      class="c2"
      href="https://simple.wikipedia.org/wiki/Renaissance_man"
      rel="nofollow noopener noreferrer external"
    >
      Renaissance Man
    </a>
     
    and try out as many different professions as possible.
  </p>
  <p
    class="c1"
  >
    These are some of the projects I created over the last years.
  </p>
</div>
`;
